modules/nifti/src/niftireader.cpp

```diff
@@ -343,37 +343,37 @@ std::shared_ptr<NiftiReader::VolumeSequence> NiftiReader::readData(const std::st
         // Transfer Function
-        auto volRAM = volumes->front()->getRepresentation<VolumeRAM>();
-        auto minmax = util::volumeMinMax(volRAM);
-        // minmax always have four components, unused components are set to zero.
-        // Hence, only consider components used by the data format
-        dvec2 dataRange(minmax.first[0], minmax.second[0]);
-        // min/max of all components
-        for (size_t component = 1; component < format->getComponents(); ++component) {
-            dataRange = dvec2(glm::min(dataRange[0], minmax.first[component]),
-                              glm::max(dataRange[1], minmax.second[component]));
-        }
-        dm.dataRange = dataRange;
-
-        if (niftiImage->scl_slope != 0) {
-            // If the scl_slope field is nonzero, then each voxel value in the dataset
-            // should be scaled as
-            //    y = scl_slope  x + scl_inter
-            // where x = stored voxel value and y = "true" voxel value
-            dm.valueRange.x = static_cast<double>(niftiImage->scl_slope) * dm.dataRange.x +
-                              static_cast<double>(niftiImage->scl_inter);
-            dm.valueRange.y = static_cast<double>(niftiImage->scl_slope) * dm.dataRange.y +
-                              static_cast<double>(niftiImage->scl_inter);
-
-        } else {
-            dm.valueRange = dm.dataRange;
+        for (auto& vol : *volumes) {
+
+            auto volRAM = vol->getRepresentation<VolumeRAM>();
+            auto minmax = util::volumeMinMax(volRAM);
+            // minmax always have four components, unused components are set to zero.
+            // Hence, only consider components used by the data format
+            dvec2 dataRange(minmax.first[0], minmax.second[0]);
+            // min/max of all components
+            for (size_t component = 1; component < format->getComponents(); ++component) {
+                dataRange = dvec2(glm::min(dataRange[0], minmax.first[component]),
+                                  glm::max(dataRange[1], minmax.second[component]));
+            }
+            dm.dataRange = dataRange;
+
+            if (niftiImage->scl_slope != 0) {
+                // If the scl_slope field is nonzero, then each voxel value in the dataset
+                // should be scaled as
+                //    y = scl_slope  x + scl_inter
+                // where x = stored voxel value and y = "true" voxel value
+                dm.valueRange.x = static_cast<double>(niftiImage->scl_slope) * dm.dataRange.x +
+                                  static_cast<double>(niftiImage->scl_inter);
+                dm.valueRange.y = static_cast<double>(niftiImage->scl_slope) * dm.dataRange.y +
+                                  static_cast<double>(niftiImage->scl_inter);
+
+            } else {
+                dm.valueRange = dm.dataRange;
+            }
+            vol->dataMap_ = dm;
```

Comment:
Setting different min max for each volume in the sequence will make it impossible to compare volumes in the sequence. And will probably lead to a lot of confusion... unless we have absolute tf this is problematic